packages/main/src/backend/import/proxyConfig.ts

```diff
@@ -0,0 +1,111 @@
+import http from 'http';
+import https from 'https';
+import { ProxyAgent } from 'proxy-agent';
+import logger from '/@/logging/logger';
+
+// Store original global agents
+let originalHttpAgent: typeof http.globalAgent | null = null;
+let originalHttpsAgent: typeof https.globalAgent | null = null;
+let currentProxyAgent: ProxyAgent | null = null;
+
+/**
+ * Gets the proxy configuration from environment variables
+ * Returns the proxy URL if found, and the NO_PROXY setting for bypass rules
+ */
+export function getProxyConfiguration(): { proxyUrl?: string; noProxy?: string } {
+  // Check common proxy environment variables
+  const proxyUrl =
+    process.env.HTTPS_PROXY ??
+    process.env.https_proxy ??
+    process.env.HTTP_PROXY ??
+    process.env.http_proxy ??
+    process.env.ALL_PROXY ??
+    process.env.all_proxy;
+
+  // Get NO_PROXY setting for bypass rules
+  const noProxy = process.env.NO_PROXY ?? process.env.no_proxy;
+
```

Comment:
Can you be more specified with this env var? I tought it means "do not use proxy", while now I understand it means maybe "skip proxy for these hosts".

Reply:
It's nice that you explained it to me, but I'm asking you to rename the environment variable now to avoid confusion for others who may have the same question.

Reply:
The environment variable name is a convention since 1994. It's not something I can easily change. 😊

Sources:
- https://superuser.com/questions/944958/are-http-proxy-https-proxy-and-no-proxy-environment-variables-standard
- https://about.gitlab.com/blog/we-need-to-talk-no-proxy/